uint32 n_cracks
Crack[] cracks